### ISSUE TITLE
BUG: AutosizeInput causes infinite loop

### ISSUE THREAD

**Issue by CWSites** [JedWatson/react-select#3290]
I'm getting a runtime error when trying to render the following.
```
let defaultOptions = [];

<Select className="react-select" classNamePrefix="react-select" options={defaultOptions} ref="filter"  />
```

Here is the output, I'm not using `AutosizeInput` on any of my elements, nor did I install it. However it is installed as a dependency of `React-Select` and shows in my `package-lock.json`
```
invariant.js:42 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at invariant (invariant.js:42)
    at scheduleWorkImpl (react-dom.development.js:12125)
    at scheduleWork (react-dom.development.js:12082)
    at Object.enqueueSetState (react-dom.development.js:6644)
    at AutosizeInput../node_modules/react/cjs/react.development.js.Component.setState (react.development.js:238)
    at AutosizeInput.updateInputWidth (AutosizeInput.js:160)
    at AutosizeInput.componentDidMount (AutosizeInput.js:100)
```

## package-lock.json
```
    "react-select": {
      "version": "2.1.2",
      "resolved": "https://registry.npmjs.org/react-select/-/react-select-2.1.2.tgz",
      "integrity": "sha512-+ceiz2KwIeEBxT/PgAXBIGohLXfa9YhkfwFSHMlqpTL55JYvjhgkGoBxoasGcMGeQ49J3RhAKZDD+x6ZHKmj6g==",
      "requires": {
        "classnames": "^2.2.5",
        "emotion": "^9.1.2",
        "memoize-one": "^4.0.0",
        "prop-types": "^15.6.0",
        "raf": "^3.4.0",
        "react-input-autosize": "^2.2.1",
        "react-transition-group": "^2.2.1"
      }
    }
```

I believe this is related to issues that were fixed in the past, https://github.com/JedWatson/react-select/issues/86, https://github.com/JedWatson/react-select/issues/567, https://github.com/JedWatson/react-select/issues/741, https://github.com/JedWatson/react-select/issues/860, https://github.com/JedWatson/react-select/issues/2636

**Comment by datner** [comment]
could you produce a codesandbox link or some other way to replicate this? I can't reproduce your error

**Comment by CWSites** [comment]
I ended up using `shouldComponentUpdate` to be very specific about when to re-render. I believe my issue is due to how the `MenuList` works. When the options load it updates the props/state of the `<Select>` component which was causing my entire component to re-render. I don't understand why state/props is being used for `onMouseOver` or `onMouseMove` but I see that it is being addressed in  #3249.
